homeassistant/components/elmax/common.py

```diff
@@ -77,11 +87,13 @@ async def _async_update_data(self):
             async with async_timeout.timeout(DEFAULT_TIMEOUT):
                 # Retrieve the panel online status first
                 panels = await self._client.list_control_panels()
-                panels = list(filter(lambda x: x.hash == self._panel_id, panels))
+                panel = next(
+                    (panel for panel in panels if panel.hash == self._panel_id), None
+                )
 
                 # If the panel is no more available within the given. Raise config error as the user must
                 # reconfigure it in order to  make it work again
-                if len(panels) < 1:
+                if not panel:
                     _LOGGER.error(
```

Comment:
We can include the message as the exception argument instead. It will be logged automatically by the config entry when handling `ConfigEntryAuthFailed`.

Reply:
DONE.
That's actually the best trade off. Thanks for the tip.